src/task/tasks.rs

```diff
@@ -315,19 +296,32 @@ impl Task {
 }
 
 extern "C" fn task_exit() {
-    panic!("Current task has exited");
+    unsafe {
+        current_task_terminated();
+    }
+    schedule();
 }
 
 #[allow(unused)]
 #[no_mangle]
 extern "C" fn apply_new_context(new_task: *mut Task) -> u64 {
     unsafe {
         let mut pt = (*new_task).page_table.lock();
-        pt.copy_entry(&this_cpu().get_pgtable(), PGTABLE_LVL3_IDX_PERCPU);
+        this_cpu().populate_page_table(&mut pt);
         pt.cr3_value().bits() as u64
     }
 }
 
+#[allow(unused)]
+#[no_mangle]
+extern "C" fn on_switch(new_task: *mut Task) {
```

Comment:
```suggestion
extern "C" fn on_switch(new_task: &mut Task) {
```

That way no unsafe is required inside the function.